WordPress/gutenberg
# Query Loop: Fix posts list variation detection

## What?
Fixes: https://github.com/WordPress/gutenberg/issues/58168

Core Query Loop's variation `posts list` was missing a proper way of being detected if is active(by not using the `isActive` API). The reason for that is that the aforementioned API was introduced later.

This PR uses the proper APIs for variations and additionally removes the `__experimentalGetMatchingVariation`(that you could say it evolved into the `getActiveBlockVariation`).

## Testing Instructions
1. Existing Query Loop blocks should remain intact
2. Insert a `Posts List` block(the variation) and observe that as long as you also preserve the queried postType to be `post`, the proper title is shown in all places(list view, block card, breadcrumbs, etc..)

## Bonus testing
1. Register a block patterns with `blockTypes: ['core/query/posts-list'] and when you insert the block in the editor click `Choose`. Your pattern should be suggested.
